src/libecalc/common/time_utils.py

```diff
@@ -53,6 +53,21 @@ def intersects(first: Period, second: Period) -> bool:
         """
         return first.start in second or second.start in first
 
+    @staticmethod
+    def intersection(first: Period, second: Period) -> Period:
+        """
+        Args:
+            first:
+            second:
+
+        Returns:
+
+        """
+        if not Period.intersects(first, second):
+            return Period()
```

Comment:
In particular here, as Period() would indicate from datetime.min to datetime.max due to defaults. I agree that None makes sense, unless there is an attempt to avoid "null", to be able to also "use the None-Period in further logic", but I cannot come up with what would make sense to have as start and end in that case.